sonarts-core/src/tools/newRule.ts

```diff
@@ -1,269 +1,269 @@
+}
+
+function replace(text: string, dictionary: any): string {
+  for (let tok in dictionary) {
+    text = text.replace(new RegExp(escapeRegExp(tok), "g"), dictionary[tok]);
+  }
+  return text;
+}
+
+function copyWithReplace(src: string, dest: string, dict: any) {
+  let content = fs.readFileSync(src).toString();
+  let newContent = replace(content, dict);
+  fs.writeFileSync(dest, newContent);
+}
+
+function createFiles(
+  rootFolder: string,
+  ruleClassName: string,
+  ruleNameDash: string,
+  ruleTitle: string,
+  rspecKey: string,
+) {
+  const templatesFolder = `${rootFolder}/sonarts-core/resources/new-rule-templates`;
+
+  let ruleMetadata: any = {};
+  ruleMetadata["___RULE_NAME_DASH___"] = ruleNameDash;
+  ruleMetadata["___RULE_CLASS_NAME___"] = ruleClassName;
+  ruleMetadata["___RULE_TITLE___"] = ruleTitle;
+  ruleMetadata["___RSPEC_KEY___"] = rspecKey;
+
+  copyWithReplace(
+    `${templatesFolder}/rule.template_ts`,
+    `${rootFolder}/sonarts-core/src/rules/${ruleClassName}.ts`,
+    ruleMetadata,
+  );
+
+  const testPath = `${rootFolder}/sonarts-core/tests/rules/${ruleClassName}`;
+
+  fs.mkdirSync(testPath);
+
+  copyWithReplace(`${templatesFolder}/unitTest.template_ts`, `${testPath}/${ruleClassName}.test.ts`, ruleMetadata);
+}
+
+function updateSonarTsJson(rootFolder: string, ruleNameDash: string) {
+  let sonarTsJson = JSON.parse(fs.readFileSync(`${rootFolder}/sonarts-core/tslint-sonarts.json`).toString());
+  sonarTsJson.rules[ruleNameDash] = true;
+  fs.writeFileSync(
+    `${rootFolder}/sonarts-core/tslint-sonarts.json`,
+    stringify(sonarTsJson, { cmp: (a, b) => (a.key < b.key ? -1 : 1), space: 2 }) + "\n",
+  );
+}
+
+function getRuleTitleAndRspecKey(rspecRuleFolder: string, rspecId: string) {
+  try {
+    const fileText = fs.readFileSync(`${rspecRuleFolder}/${rspecId}.json`).toString();
+
+    let ruleData = JSON.parse(fileText);
+    return { ruleTitle: ruleData["title"], rspecKey: ruleData["ruleSpecification"] };
+  } catch (err) {
+    console.error("could not find metadata for the rule. Have you run the rspec-api (with correct parameters)?");
+    showHelp();
+    throw err;
+  }
+}
+
+function getDescription(rspecRuleFolder: string, rspecId: string) {
+  try {
+    let replaceDict: any = {};
+    replaceDict["<em>"] = "";
+    replaceDict["</em>"] = "";
+    replaceDict["<p>"] = "";
+    replaceDict["</p>"] = `\n`;
+    replaceDict["<code>"] = "`";
+    replaceDict["</code>"] = "`";
+    replaceDict["<pre>"] = "```typescript";
+    replaceDict["</pre>"] = "```";
+    replaceDict["<h2>"] = "## ";
+    replaceDict["</h2>"] = `\n`;
+
+    const fileText = fs.readFileSync(`${rspecRuleFolder}/${rspecId}.html`).toString();
+    return replace(fileText, replaceDict);
+  } catch (err) {
+    console.error("could not find metadata for the rule. Have you run the rspec-api?");
+    showHelp();
+    throw err;
+  }
+}
+
+function updateReadme(rootFolder: string, ruleTitle: string, ruleNameDash: string) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: The function updateReadme has a cyclomatic complexity of 17 which is higher than the threshold of 10 ([typescript:S1541](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1541))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=stas%2Fconsistent-file-names&issues=AWDB3HvcqDrCo5ysgyxV&open=AWDB3HvcqDrCo5ysgyxV)

[ISSUE_KEY]: AWDB3HvcqDrCo5ysgyxV

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED